### PR TITLE
Fix + test for renames

### PR DESCRIPTION
**Pull request by icarus-sparry**
Not sure what people think of putting in binary files for the bsdiff.

Factored as 3 patches, updated swupdlib.bash functionality, a test for rename, and the fix for rename.

### CODE REVIEW

**Comment by matthewrsj** [comment]
You seem to be checking in a couple binary delta files, but it looks like those are automatically added during the test setup?

**Comment by icarus-sparry** [comment]
I don't believe they are generated during the test setup, they are copied from web-dir to staged. We definitely could do with some better tooling around creating tests.

**Comment by matthewrsj** [comment]
> We definitely could do with some better tooling around creating tests.

Agree, see #258 

Instead of adding the delta packs, you can just add the files to the `web-dir/20/staged` directory.

I also noticed you aren't checking the output with a lines-checked file.

**Comment by icarus-sparry** [comment]
I think that checking the output with a "lines-checked" adds very little value, and just adds to future maintenance issues. The test checks that the files which should be renamed are removed, and the new ones are present, which is the thing we are trying to test, not that we are outputting the correct copyright message.
OK, will rework the test to not use the delta files but have staged ascii files instead.

**Comment by icarus-sparry** [comment]
If we don't have the delta files, how do we know that we are actually doing the rename, rather than just doing normal processing of adding files from staging and removing them from target-dir?

**Comment by matthewrsj** [comment]
@icarus-sparry let's stick with deltas

**Comment by matthewrsj** [comment]
lgtm and the test gives me confidence that this is actually working. I would like one more +1 on this before merging though.

**Comment by matthewrsj** [comment]
@phmccarty review ping

File: test/functional/swupdlib.bash [hunk 1]
```diff
@@ -107,4 +121,30 @@ sign_manifest_mom() {
...
+  while read -u 9 line
+  do  [ -z "$line" ] && break
+  done
+  # Process lines
+  mkdir -p "$DIR/target-dir"
+  while read -u 9 code hash ver path
+  do
+      case "$code" in
+	  ("F"*) sudo cp "$DIR/web-dir/$REL/staged/$hash" "$DIR/target-dir/$path" ;;
+	  ("D"*) sudo mkdir -p "$DIR/target-dir/$path" ;;
```
Comment (phmccarty) [inline test/functional/swupdlib.bash]: Do you want to add symlink ("L") support right now, or defer to later?

Comment (icarus-sparry) [reply test/functional/swupdlib.bash]: I propose to defer it to later.

Comment (phmccarty) [reply test/functional/swupdlib.bash]: Okay, I opened #302 to track this.

Comment (phmccarty) [reply test/functional/swupdlib.bash]: +1 to defer for now

**Pull request merged**